### PR TITLE
Apply timeLimitMultiplier to datatype methods, not just class members.

### LINKED ISSUE #1231: timeLimitMultiplier ignored on datatype members
In the following example, the `timeLimitMultiplier` annotation has no effect on the Boogie output.  This is due to the implementation only looking for the `timeLimitMultiplier` annotations in class members.  I'll open a PR with a fix after filing this.
```
datatype Dt = Dt(x:int)
{
  method {:timeLimitMultiplier 5} Test(y:int) {
    var z := y;
    assert z == y;
  }
}
```

### PR DESCRIPTION
**Pull request by parno**
Closes #1231.

### CODE REVIEW

File: Source/Dafny/Rewriter.cs [hunk 1]
```diff
@@ -1498,9 +1498,15 @@ public TimeLimitRewriter(ErrorReporter reporter)
 
     internal override void PreResolve(ModuleDefinition m) {
       foreach (var d in m.TopLevelDecls) {
-        if (d is ClassDecl) {
-          var c = (ClassDecl)d;
```
Comment (RustanLeino) [inline Source/Dafny/Rewriter.cs:1502]: The proposed fix still misses `codatatype` and `newtype` declarations. Go back to the original code and just change `ClassDecl` on L1501 and L1502 to `TopLevelDeclWithMembers`.

File: Test/git-issues/git-issue-1231.dfy [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+datatype Dt = Dt(x:int)
```
Comment (RustanLeino) [inline Test/git-issues/git-issue-1231.dfy:4]: While you're at it, please also add a similar test for `codatatype` and `newtype`.

**Review by RustanLeino — CHANGES_REQUESTED** [review]

File: Source/Dafny/Rewriter.cs (hunk 1 above)
Comment (parno) [reply Source/Dafny/Rewriter.cs:1502]: Thanks for the pointer to `TopLevelDeclWithMembers`.  That does look like a cleaner solution.

File: Test/git-issues/git-issue-1231.dfy (hunk 2 above)
Comment (parno) [reply Test/git-issues/git-issue-1231.dfy:4]: Done.

**Review by RustanLeino — APPROVED** [review]

**Pull request merged**